File: pulsar-tools/pulsar-browser/src/main/java/com/github/kklisura/cdt/protocol/v2023/events/page/FrameNavigated.java
package com.github.kklisura.cdt.protocol.v2023.events.page;

/*-
 * #%L
 * cdt-java-client
 * %%
 * Copyright (C) 2018 - 2023 Kenan Klisura
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

import com.github.kklisura.cdt.protocol.v2023.support.annotations.Experimental;
import com.github.kklisura.cdt.protocol.v2023.types.page.Frame;
import com.github.kklisura.cdt.protocol.v2023.types.page.NavigationType;

/**
 * Fired once navigation of the frame has completed. Frame is now associated with the new loader.
 */
public class FrameNavigated {

  private Frame frame;

  @Experimental
  private NavigationType type;

  /** Frame object. */
  public Frame getFrame() {
    return frame;
  }

  /** Frame object. */
  public void setFrame(Frame frame) {
    this.frame = frame;
  }

  public NavigationType getType() {
    return type;
  }

  public void setType(NavigationType type) {
    this.type = type;
  }
}

File: pulsar-core/pulsar-tools/pulsar-browser/src/main/kotlin/ai/platon/cdt/kt/protocol/events/page/FrameNavigated.kt
@file:Suppress("unused")
package ai.platon.cdt.kt.protocol.events.page

import ai.platon.cdt.kt.protocol.support.annotations.Experimental
import ai.platon.cdt.kt.protocol.types.page.Frame
import ai.platon.cdt.kt.protocol.types.page.NavigationType
import com.fasterxml.jackson.`annotation`.JsonProperty

/**
 * Fired once navigation of the frame has completed. Frame is now associated with the new loader.
 */
data class FrameNavigated(
  @param:JsonProperty("frame")
  val frame: Frame,
  @param:JsonProperty("type")
  @param:Experimental
  val type: NavigationType,
)
